[.NET 10] CLI - Creating new MAUI with Sample using CLI fails to restore with NU1107 error

### Description

Creating new MAUI Sample App fails to restore with NU1107 error

### Steps to Reproduce

1. Install the .NET 10 GA from [here](https://dev.azure.com/dnceng/internal/_build/results?buildId=2823170&view=artifacts&type=publishedArtifacts)  or windows: [here](https://artprodcus3.artifacts.visualstudio.com/Ab55de4ed-4b5a-4215-a8e4-0a0a5f71e7d8/7ea9116e-9fac-403d-b258-b31fcf1bb293/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy9wcm9qZWN0SWQvN2VhOTExNmUtOWZhYy00MDNkLWIyNTgtYjMxZmNmMWJiMjkzL2J1aWxkSWQvMjgyMzE3MC9hcnRpZmFjdE5hbWUvYXNzZXRz0/content?format=file&subPath=%2Fshipping%2Fassets%2FSdk%2F10.0.100-rtm.25523.104%2Fdotnet-sdk-10.0.100-win-x64.exe) and mac: [here](https://artprodcus3.artifacts.visualstudio.com/Ab55de4ed-4b5a-4215-a8e4-0a0a5f71e7d8/7ea9116e-9fac-403d-b258-b31fcf1bb293/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy9wcm9qZWN0SWQvN2VhOTExNmUtOWZhYy00MDNkLWIyNTgtYjMxZmNmMWJiMjkzL2J1aWxkSWQvMjgyMzE3MC9hcnRpZmFjdE5hbWUvYXNzZXRz0/content?format=file&subPath=%2Fshipping%2Fassets%2FSdk%2F10.0.100-rtm.25523.104%2Fdotnet-sdk-10.0.100-osx-arm64.pkg). Alternatively, you can use the [PR Build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=12649344&view=results) for windows validation.
2. Install MAUI workload from the rollback file along with the necessary feeds: [rollback.json](https://github.com/user-attachments/files/23121293/rollback.json)
3. Create an empty folder and open it on terminal/CMD
4. Run the command `dotnet new maui --sample-content`
5. Project fails to be restored with 2 warnings and 1 error.

**Actual Behavior:**
Project fails to be restored with 2 warnings and 1 error.
```
Restoring /Users/vstester/Desktop/untitled folder 3/sample/sample.csproj:
/Users/vstester/Desktop/untitled folder 3/sample/sample.csproj : warning NU1608: Detected package version outside of dependency constraint: CommunityToolkit.Maui 11.1.1 requires Microsoft.Maui.Controls (>= 9.0.30 && < 10.0.0) but version Microsoft.Maui.Controls 10.0.0 was resolved.
/Users/vstester/Desktop/untitled folder 3/sample/sample.csproj : warning NU1608: Detected package version outside of dependency constraint: CommunityToolkit.Maui.Core 11.1.1 requires Microsoft.Maui.Core (>= 9.0.30 && < 10.0.0) but version Microsoft.Maui.Core 10.0.0 was resolved.
/Users/vstester/Desktop/untitled folder 3/sample/sample.csproj : error NU1107: Version conflict detected for Microsoft.Maui.Essentials. Install/reference Microsoft.Maui.Essentials 10.0.0 directly to project sample to resolve this issue .sample -> Microsoft.Maui.Controls 10.0.0 -> Microsoft.Maui.Controls.Core 10.0.0 -> Microsoft.Maui.Core 10.0.0 -> Microsoft.Maui.Essentials (>= 10.0.0) sample -> CommunityToolkit.Maui 11.1.1 -> CommunityToolkit.Maui.Core 11.1.1 -> Microsoft.Maui.Essentials (>= 9.0.30 && < 10.0.0).
Restore failed.
Post action failed.
```

https://github.com/user-attachments/assets/b4b805a6-ae89-43e8-94dd-538878e2c51f

**Expected Behavior:**
Project can be successfully created without any warnings or errors.

### Version with bug

.NET SDK 10.0.100

### Is this a regression from previous behavior?

Yes, this is a regression. The issue cannot be repro on .NET SDK version 10.0.100-rc.2.25502.107.

### Affected platforms

Windows, macOS

--- a/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj
+++ b/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj
@@ -19,8 +19,6 @@
 		<ImplicitUsings>enable</ImplicitUsings>
 		<Nullable>enable</Nullable>
 <!--#if (IncludeSampleContent) -->
-		<!-- https://github.com/CommunityToolkit/Maui/issues/2921 -->
-		<NoWarn>NU1608</NoWarn>
 		<MauiEnableXamlCBindingWithSourceCompilation>true</MauiEnableXamlCBindingWithSourceCompilation>
 <!--#endif -->
 

--- a/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj
+++ b/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj
@@ -81,7 +81,7 @@
 <!--#if (IncludeSampleContent) -->
 	<ItemGroup>
 		<PackageReference Include="CommunityToolkit.Mvvm" Version="8.3.2" />
-		<PackageReference Include="CommunityToolkit.Maui" Version="11.1.1" />
+		<PackageReference Include="CommunityToolkit.Maui" Version="12.3.0" />
 		<PackageReference Include="Microsoft.Data.Sqlite.Core" Version="8.0.8" />
 		<PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.10" />
 		<PackageReference Include="Syncfusion.Maui.Toolkit" Version="1.0.7" />

--- a/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj
+++ b/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj
@@ -74,7 +74,6 @@
 
 	<ItemGroup>
 		<PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
-		<PackageReference Include="Microsoft.Maui.Essentials" Version="$(MauiVersion)" />
 		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="MS_EXT_LOG_DEBUG_VERSION" />
 	</ItemGroup>
 

--- a/src/TestUtils/src/Microsoft.Maui.IntegrationTests/SimpleTemplateTest.cs
+++ b/src/TestUtils/src/Microsoft.Maui.IntegrationTests/SimpleTemplateTest.cs
@@ -40,16 +40,6 @@ public class SimpleTemplateTest : BaseTemplateTests
 				"</Project>",
 				"<PropertyGroup><Version>1.0.0-preview.1</Version></PropertyGroup></Project>");
 
-		string[]? warningsToIgnore = null;
-
-		if (additionalDotNetNewParams.Contains("sample-content", StringComparison.OrdinalIgnoreCase))
-		{
-			warningsToIgnore = new string[]
-			{
-				"NU1608", // https://github.com/CommunityToolkit/Maui/issues/2921
-			};
-		}
-
 		// We only have these packs for Android
 		if (additionalDotNetBuildParams.Contains("UseMonoRuntime=false", StringComparison.OrdinalIgnoreCase))
 		{
@@ -64,7 +54,7 @@ public class SimpleTemplateTest : BaseTemplateTests
 		}
 
 		string target = shouldPack ? "Pack" : "";
-		Assert.IsTrue(DotnetInternal.Build(projectFile, config, target: target, properties: buildProps, msbuildWarningsAsErrors: true, warningsToIgnore: warningsToIgnore),
+		Assert.IsTrue(DotnetInternal.Build(projectFile, config, target: target, properties: buildProps, msbuildWarningsAsErrors: true),
 			$"Project {Path.GetFileName(projectFile)} failed to build. Check test output/attachments for errors.");
 	}
 

